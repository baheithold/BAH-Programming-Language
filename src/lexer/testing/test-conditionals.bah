# This file tests valid conditionals


if (num == 0) {
    return 0;
}
else if (isOdd?(num)) {
    return 1;
}
else {
    return 2;
}
