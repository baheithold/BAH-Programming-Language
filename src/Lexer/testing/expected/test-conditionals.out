IF_KEYWORD
OPAREN
ID_TYPE num
EQUALS_COMPARATOR
INTEGER_TYPE 0
CPAREN
OBRACE
RETURN_KEYWORD
INTEGER_TYPE 0
SEMICOLON
CBRACE
ELSE_KEYWORD
IF_KEYWORD
OPAREN
ID_TYPE isOdd?
OPAREN
ID_TYPE num
CPAREN
CPAREN
OBRACE
RETURN_KEYWORD
INTEGER_TYPE 1
SEMICOLON
CBRACE
ELSE_KEYWORD
OBRACE
RETURN_KEYWORD
INTEGER_TYPE 2
SEMICOLON
CBRACE
END_OF_INPUT
